src/library/export/trackexportworker.cpp

```diff
@@ -5,33 +5,98 @@
 #include <QMessageBox>
 #include <QDebug>
 
+namespace {
+
+// Iterate over a list of tracks and generate a minimal set of files to copy.
+// Finds duplicate filenames.  Munges filenames if they refer to different files,
+// and skips if they refer to the same disk location.  Returns a map from
+// QString (the destination possibly-munged filenames) to QFileinfo (the source
+// file information).
+QMap<QString, QFileInfo> createCopylist(const QList<TrackPointer>& tracks) {
+    // QMap is a non-obvious return value, but it's easy for callers to use
+    // in practice and is the best object for producing the final list
+    // efficiently.
+    QMap<QString, QFileInfo> copylist;
+    for (const auto& it : tracks) {
+        auto fileinfo = it->getFileInfo();
+
+        // The munging loop is on the outside because for each munged name
+        // we have to see if we already munged the same file the same way.
+        bool success = false;
+        for (int i = 0; i < 10000; ++i) {
+            QString dest_filename;
+            // For the first case, just use the filename as-is.
+            if (i == 0) {
+                dest_filename = fileinfo.fileName();
+            } else {
+                // We don't have total control over the inputs, so definitely
+                // don't use .arg().arg().arg().
+                const QString index_str =
+                        QString("%1").arg(i, 4, 10, QChar('0'));
+                dest_filename = QString("%1-%2.%3")
+                        .arg(fileinfo.baseName(), index_str,
+                             fileinfo.completeSuffix());
+            }
+            auto seen_it = copylist.find(dest_filename);
+            if (seen_it == copylist.end()) {
+                // Usual case -- haven't seen this filename before, so add it.
+                copylist[dest_filename] = fileinfo;
+                success = true;
+                break;
+            }
+
+            if (fileinfo == *seen_it) {
+                // These are the same file, so don't add this new one to the
```

Comment:
The docs seem to say otherwise:
http://doc.qt.io/qt-4.8/qfileinfo.html#operator-eq-eq
```
Warning: This will not compare two different symbolic links pointing to the same file.
```